tests/ott/go/test/polymorphism_test.go

```diff
@@ -0,0 +1,157 @@
+package test
+
+import (
+	"encoding/json"
+	"regexp"
+	"testing"
+
+	"github.com/kaltura/KalturaOttGeneratedAPIClientsGo/kalturaclient/enums/channelorderby"
+	"github.com/kaltura/KalturaOttGeneratedAPIClientsGo/kalturaclient/enums/groupbyorder"
+	"github.com/kaltura/KalturaOttGeneratedAPIClientsGo/kalturaclient/types"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestPolySerialization(t *testing.T) {
+	t.Parallel()
+	channelFilter := GetChannelFilter()
+	_ = GetAssets(t, channelFilter)
+}
+
+func GetAssets(t *testing.T, filter types.AssetFilterInterface) []types.AssetContainer {
+	// request
+	bytes, err := json.Marshal(&filter)
+	assert.NoError(t, err)
+	space := regexp.MustCompile(`\s+`)
+	channelFilterString := space.ReplaceAllString(GetChannelFilterString(), "")
+	assert.Equal(t, channelFilterString, string(bytes))
```

Comment:
if you want to compare jsons correctly
```
There is a popular testing package in golang called require github.com/stretchr/testify/require and it will do this for you.

func TestJsonEquality(t *testing.t) { 
  expected := `{"a": 1, "b": 2} `
  actual := ` {"b":   2, "a":   1}`
  require.JSONEq(t, expected, actual)
}


GoDocs: https://godoc.org/github.com/stretchr/testify/require#JSONEqf
```